packages/server/src/constraints.ts

```diff
@@ -12,40 +12,55 @@ const MAX_SERVICE_TYPE_LENGTH = 256
 const MAX_SERVICE_ENDPOINT_LENGTH = 512
 const MAX_ID_LENGTH = 32
 
-export function assertValidIncomingOp(
-  op: unknown,
-): asserts op is plc.OpOrTombstone {
-  const byteLength = cborEncode(op).byteLength
+export function validateIncomingOp(input: unknown): plc.OpOrTombstone {
+  const byteLength = cborEncode(input).byteLength
   if (byteLength > MAX_OP_BYTES) {
     throw new ServerError(
       400,
       `Operation too large (${MAX_OP_BYTES} bytes maximum in cbor encoding)`,
     )
   }
-  if (!check.is(op, plc.def.opOrTombstone)) {
-    throw new ServerError(400, `Not a valid operation: ${JSON.stringify(op)}`)
+
+  // We *need* to parse, and use the result of the parsing, to ensure that any
+  // unknown fields are removed from the input. "@atproto/common"'s check
+  // function will not remove unknown fields.
+  const result = plc.def.opOrTombstone.safeParse(input)
```

Comment:
If we stripped some fields off, this should error because the signature will no longer match the data. But it may be nice to proactively return an error here notifying the sender that there were unspecified fields on the operation

Reply:
I wonder if it would instead be better to make all of the zod schemas `strict` (https://zod.dev/?id=strict) 🤔

Reply:
This would mean changing `@did-plc/lib`. I'd be down with that but just wanna make 100% sure that's what we want.

Reply:
hmm good point - and i guess `@did-plc/lib` would then say events that are already in the directory are invalid 🤔 

Alright let's go ahead and just roll with this for now